src/components/menu-surface/menu-surface.tsx

```diff
@@ -163,4 +167,24 @@ export class MenuSurface {
             this.dismiss.emit();
         }
     }
+
+    private ensureMenuFitsInViewPort() {
+        setTimeout(() => {
+            const vh = Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);
+            const { top, bottom } = this.host.getBoundingClientRect();
+            if (vh > bottom && top > 0) {
+                // The surface is rendered inside the viewport :)
+                return;
+            }
+
+            // We don't know if the surface will be rendered "upwards" or "downwards".
+            // We therefore calculate the space above or below the surface and subtract
+            // that from its height.
+            const spaceAboveSurface = Math.max(top, 0);
+            const spaceBelowSurface = Math.max(vh - bottom, 0);
+            const extraCosmeticSpace = 16;
+            const maxHeight = vh - Math.min(spaceAboveSurface, spaceBelowSurface) - extraCosmeticSpace;
+            this.host.style['height'] = `${maxHeight}px`
```

Comment:
I'm not entirely sure I understand what this is doing, or trying to do.

Because of the `if` on line 175, if we get here, either `spaceAboveSurface` or `spaceBelowSurface` must be 0, which gives us a constant `maxHeight` of `vh - extraCosmeticSpace`.

It seems like we're meaning to consider the placement of the menu-trigger on the page, and crop the height of the menu to the remaining space above or below?